### PR TITLE
Mm/groups

### PR DESCRIPTION
**Pull request by MarchandMD**
# What does this PR do? 
- adds endpoint to GET `/groups`
- adds endpoint to GET `/groups/:id`

# How does this PR accomplish that?
- see notes in git diff

### CODE REVIEW

File: dist/models/Players.js [hunk 1]
```diff
@@ -3,23 +3,24 @@
...
+    type: String,
+    required: false,
+    default: 'email@email.com'
   },
   group_id: {
-    type: Number,
+    type: String,
     required: false
   }
 });
```
Comment (MarchandMD) [inline dist/models/Players.js:26]: this is the result of running `npm run build`

File: dist/models/Tasks.js [hunk 2]
```diff
@@ -20,4 +22,4 @@ const TaskSchema = mongoose.Schema({
     required: false
   }
 });
-module.exports = mongoose.model("Tasks", TaskSchema);
```
Comment (MarchandMD) [inline dist/models/Tasks.js:23]: this is the result of running npm run build

File: dist/routes/groups.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ *    parameters:
+ *    - name: groupId
+ *      in: path
+ *      required: true
+ *      schema:
+ *        type: string
+ * */
+groupRouter.get("/:id", (request, response) => {
+  Groups.findById(request.params.id).then(resp => response.status(200).json(resp)).catch(err => response.status(400).json("Request Failed"));
+});
```
Comment (MarchandMD) [inline dist/routes/groups.js:47]: this is the result of running npm run build

File: dist/routes/tasks.js [hunk 4]
```diff
@@ -119,7 +119,7 @@ router.patch("/:id", (request, response) => {
  * /tasks/{taskId}:
  *  delete:
  *    tags:
- *      - [task/:id]
+ *      - [tasks/:id]
  *    summary: Use this endpoint to delete a single task
  *    description: if you have a task id, then include it in the http request, and it will be removed from the database; no params required
  *    responses:
```
Comment (MarchandMD) [inline dist/routes/tasks.js:125]: this is the result of running npm run build

File: src/app.js [hunk 5]
```diff
@@ -14,10 +15,12 @@ app.use(
 
 const taskRouter = require('./routes/tasks');
 const playerRouter = require('./routes/players');
+const groupRouter = require('./routes/groups');
```
Comment (MarchandMD) [inline src/app.js:18]: adds the `groupRouter`

File: src/app.js [hunk 6]
```diff
@@ -14,10 +15,12 @@ app.use(
...
 
 app.use(express.json())
 app.use("/tasks", taskRouter)
 app.use("/players", playerRouter)
+app.use("/groups", groupRouter)
```
Comment (MarchandMD) [inline src/app.js:23]: adds the endpoint `/groups`

File: src/models/Groups.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+        type: String,
+        required: true,
+    },
+    description: {
+        type: String,
+        required: false,
+    },
+})
+
+module.exports = mongoose.model("Groups", GroupSchema)
```
Comment (MarchandMD) [inline src/models/Groups.js:14]: the new `groups` model

File: src/routes/groups.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+ *      schema:
+ *        type: string
+ * */
+groupRouter.get("/:id", (request, response) => {
+    Groups.findById(request.params.id)
+        .then((resp) => response.status(200).json(resp))
+        .catch((err) => response.status(400).json("Request Failed"))
+})
+
+module.exports = groupRouter
```
Comment (MarchandMD) [inline src/routes/groups.js:51]: comments are for Swagger documentation

**Review by msakr21 — APPROVED** [review]
Everything looks good. Made the edit to the CORS allow list localhost port as per our slack exchange so it would allow Jordan to use both repos in their own respective development servers with the FE being able to call on the BE.

File: dist/models/Players.js (hunk 1 above)
Comment (MarchandMD) [reply dist/models/Players.js:26]: The manual change happened in `src/models/Players.js`....

Then after running `npm run build` the `dist/` directory was updated

So to answer your question: it was a manual change in `src/` that initiated the change in `dist/` after running `npm run build`

Unless i'm misunderstanding the flow of ES6 -> ES5

**Pull request merged**